### DISCUSSION TITLE
How can I give the startup script access to an external GitHub repo that I have access to?

### DISCUSSION THREAD

**Discussion by ryansburgoyne** [coder/coder#9118] [General]
I have configured a GitHub app and installed it in my organization per [the docs](https://coder.com/docs/v2/latest/admin/git-providers#github-app). This way, users can authenticate to GitHub before creating a workspace, and the startup script is able to use their credentials to clone repositories owned by my organization.

However, my GitHub account has access to a third-party repository (Epic Games Unreal Engine in this case) that I want to be able to clone from the startup script as well. I was hoping that when I authenticated with GitHub, it would allow me to grant access to additional repositories (like it does in the OAuth flow), but that doesn't appear to be the case. I can only access the ones for my organization (I assume because Epic Games hasn't installed my app in their GitHub organization).

I can get it to work by adding my Coder SSH key to my GitHub account and cloning using SSH, but I was hoping to make this process more streamlined using the Git Providers. Is there any way to accomplish what I am trying to do?

**Comment by matifali** [comment]
#9121 might be related.

**Reply by ryansburgoyne** [reply]
This is definitely related and would be a step in the right direction. But I'm wondering if there is any way to make this work in the case when I definitely will not be able to install the GitHub app into the organization for the repository, but my personal account has access (e.g. Unreal Engine). It seems like the OAuth flow allows me to give access to any organizations that I have access to. Could that be used to allow cloning as well?
